File: truevfs-ext/truevfs-ext-logging/src/main/scala/net/java/truevfs/ext/logging/LogInputStream.scala
/*
 * Copyright (C) 2005-2012 Schlichtherle IT Services.
 * All rights reserved. Use is subject to license terms.
 */
package net.java.truevfs.ext.logging

import java.io._
import net.java.truecommons.io._
import net.java.truevfs.kernel.spec.cio._
import org.slf4j._

/**
 * @author Christian Schlichtherle
 */
private final class LogInputStream(
  override val origin: InputSocket[_ <: Entry],
  in: InputStream
) extends DecoratingInputStream(in) with LogCloseable with Immutable {
  override def logger = LogInputStream.logger
}

private object LogInputStream {
  private val logger = LoggerFactory.getLogger(classOf[LogInputStream])
}

File: truevfs-ext/truevfs-ext-logging/src/main/java/net/java/truevfs/ext/logging/LogInputStream.java
package net.java.truevfs.ext.logging;

import net.java.truecommons.cio.Entry;
import net.java.truecommons.cio.IoSocket;
import net.java.truecommons.io.DecoratingInputStream;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.io.InputStream;

final class LogInputStream extends DecoratingInputStream implements LogCloseable {

    private static final Logger logger = LoggerFactory.getLogger(LogInputChannel.class);

    private final IoSocket<? extends Entry> context;

    LogInputStream(final IoSocket<? extends Entry> context, final InputStream in) {
        super(in);
        this.context = context;
        opening();
    }

    @Override
    public IoSocket<? extends Entry> context() {
        return context;
    }

    @Override
    public Logger logger() {
        return logger;
    }
}
